src/serializer/ResidualFunctions.js

```diff
@@ -258,6 +263,9 @@ export class ResidualFunctions {
         funcNodes.set(functionValue, funcNode);
       };
 
+      let rewrittenBody = rewrittenAdditionalFunctions.get(instances[0].functionValue);
+      if (rewrittenBody) funcBody = t.blockStatement(rewrittenBody);
+
       if (shouldInline || instances.length === 1 || usesArguments) {
```

Comment:
This `if` statement here decides whether a function should be treated in the naive way.
Instead of that requirement above that all instances of a function must be either prepacked or not, I think you should be able to just deal with the relevant instances right here, and then always go into the native first case.